tests/SocketTest.php

```diff
@@ -267,4 +267,25 @@ public function setExpectedException($exception, $message = '', $code = 0)
             parent::setExpectedException($exception, $message, $code);
         }
     }
+
+    /**
+     * @depends testServerNonBlocking
+     */
+    public function testAssertAliveAfterClose(Socket $server)
+    {
+        $errorHandlerCalled = false;
+        try {
+            set_error_handler(function() use (&$errorHandlerCalled) {
+                $errorHandlerCalled = func_get_args();
+            });
+            $server->close();
+            $server->assertAlive();
+        } catch (Exception $e) {
+            $this->assertFalse($errorHandlerCalled, 'we do not want to trigger any warnings etc.');
+            $this->assertSame(0, $e->getCode());
+            $this->assertSame('Socket error: not a valid resource', $e->getMessage());
+        } finally {
+            restore_error_handler();
+        }
```

Comment:
Thanks for adding a test case! The `finally` block is not supported on legacy PHP versions, so perhaps you can avoid this?